content/blog/the-importance-of-ubiquitous-language/index.md

```diff
@@ -0,0 +1,181 @@
+
+Let's use an example to illustrate what we are talking about.  Say that our team is writing a shopping application and our product manager have a user interview with someone from the business and write a single story that says something along the lines of:
+
+> As a customer, I want to be able to add products to my shopping cart, so that I can continue shopping until I'm ready to check out.
+
+The architect may then make some high level architecture decisions:
+* A shopping cart should be implemented as a list of items that are comprised of a product and quantity
+* A customer is an authenticated user
+* Once the customer checks out, the items in the shopping cart should be removed
+
+Now, being in the hands of the developer, they write something like this:
+
+```java
+class Item {
+    Product product;
+    int quantity;
+}
+
+class ItemList {
+    List<Item> itemList;
+    
+    void add(Item item) {
+        //add to list
+    }
+    
+    void remove(Item item) {
+        //remove from list
+    }
+    
+    void removeAll() {
+        //removes all items from list
+    }
+}
+```
+
+Although the implementation is technically correct, the terminology they used when writing the code has now given the developers a different vocabulary than other members of the team.  Now we have introduced some translations that need to happen:
+
+* Product <--> Item
+* Cart <--> ItemList
+* ??? <--> Product
+* "Checkout" <--> "Remove all items from cart"
+
+What has been created here is a translation layer between the PM and the engineers, and worse, between engineers and the business. Requiring these translations will reduce efficiency of conversation, increase cognitive load, and possibly even break down trust between the engineering team and stakeholders.
+
+In order to streamline this communication as much as possible we need to remove all unnecessary translations.  Remember, the word ubiquitous means "found everywhere", so lets see how this should best be accomplished in a software product on a balanced team.
+
+## Usage
+In order for this language to be **found everywhere** it's important that it is defined and understood **by everyone**.  The best and easiest way to accomplish this is to ensure that all disciplines are involved in as much user-facing communication as is reasonable to stay productive.  It's not necessary that the entire development team for your application sit in every single user-interview, however on the other hand it is not of value for the development team to be involved in any.  Translation and context sharing has a cost, and one that should not be ignored.  Engineers, especially more senior ones, should know the business domain as well as the PM and designers.
```

Comment:
I've read the second sentence for line 101 and I'm not sure what it means as written.